stack.hpp

```diff
@@ -74,8 +58,16 @@ class Stack {
             throw std::out_of_range("Stack is empty");
             return std::move(storage_[0]);
         }
-}
+        }
 
+    ~Stack()
+    {
+        if (storage_ != NULL)
+        {
+            delete storage_;
```

Comment:
Wrong delete, it is an array after all.

FYI: If you call delete with a NULL pointer it does just nothing, so the if is not required.